porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -69,6 +70,36 @@ func (r *PackageVariantReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	if !pv.ObjectMeta.DeletionTimestamp.IsZero() {
+		// This object is being deleted, so we need to make sure the packagerevisions owned by this object
+		// are deleted. Normally, garbage collection can handle this, but we have a special case here because
+		// (a) we cannot delete published packagerevisions and instead have to propose deletion of them
+		// (b) we may want to orphan packagerevisions instead of deleting them.
+		for _, pr := range prList.Items {
+			if r.hasOurOwnerReference(pv, pr.OwnerReferences) {
+				r.deleteOrOrphan(ctx, &pr, pv)
+				if porchapi.LifecycleIsPublished(pr.Spec.Lifecycle) {
```

Comment:
> Does this take advantage of the fact that if a packagerevision still has lifecycle Published after the call to deleteOrOrphan, it must have deletionPolicy Orphan?

It did not, thanks for the suggestion. I have updated it so that it does, and clarified with a comment. This is needed because packagerevisions that are owned by this `PackageVariant` will be automatically deleted when the PackageVariant object is deleted. Since we don't want published package revisions to be deleted, we orphan them first to prevent the automatic deletion.